packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -123,6 +124,14 @@
       return this.queryPreviewResults?.results;
     }
 
+    /**
+     * It injects the provided {@link FeatureLocation} of the selected query in the search request.
+     *
+     * @internal
+     */
+    @Inject()
+    public location!: FeatureLocation;
```

Comment:
`location` might be undefined, change the `!` to `?`

```suggestion
    public location?: FeatureLocation;
```